spec/ravelin/event_spec.rb

```diff
@@ -175,19 +175,125 @@ class MyDummyClass < Ravelin::RavelinObject
       end
     end
 
+    context 'v2 login event' do
+      it 'throws ArgumentError with missing payload parameters' do
+        expect {
+          described_class.new(
+              name: :login,
+              payload: { }
+          )
+        }.to raise_exception(
+                 ArgumentError,
+                 /payload missing parameters: username, success, authentication_mechanism/
+        )
+      end
+
+      it 'is executed cleanly with required payload parameters' do
+        expect {
+          described_class.new(
+            name: :login,
+            payload: {
+              username: "big.jim@deliveroo.invalid",
+              customer_id: 123,
+              success: true,
+              authentication_mechanism: {
+                password: {
+                  password: "lol",
+                  success: true
+                }
+              }
+            }
+          )
+        }.to_not raise_exception
+      end
+
+      it 'it generates the correct serializable hash' do
+        output = described_class.new(
+            name: :login,
+            payload: {
+              username: "big.jim@deliveroo.invalid",
+              customer_id: 123,
+              success: true,
+              authentication_mechanism: {
+                password: {
+                  password: "lol",
+                  success: true
+                }
+              }
+            },
+            timestamp: 1586283630
+        ).serializable_hash
+
+        expect(output).to eq(
+          {
+            "authenticationMechanism" => {
+              "password" => {
+                "passwordHashed" => "07123e1f482356c415f684407a3b8723e10b2cbbc0b8fcd6282c49d37c9c1abc",
+                "success" => true
+              }
+            },
+            "customerId" => "123",
+            "success" => true,
+            "username" => "big.jim@deliveroo.invalid",
+            "timestamp" => 1586283630
+          }
+        )
+      end
+
+      it 'includes the correct event name' do
+        expect(
+          described_class.new(
+            name: :login,
+            payload: {
+              username: "big.jim@deliveroo.invalid",
+              customer_id: 123,
+              success: true,
+              authentication_mechanism: {
+                password: {
+                  password: "lol",
+                  success: true
+                }
+              }
+            }
+          ).name).to eq(:login)
+      end
+
+    end
+
     context 'ato login' do
```

Comment:
What's the difference between a v2 login event and an ATO login? Does this refer to V3?

Reply:
The V3 login event contains the same data as a V2 login endpoint, albeit with that data wrapped in a hash under a `login` key, and has a Unix timestamp with millisecond accuracy.

The V2 login event has a Unix timestamp with second level accuracy.

/shrug

Reply:
Maybe we can call this 'V3 login event' to be consistent with the 'V2 login event' naming and easier to understand!

Thanks for the explanation!